RecoTracker/TkTrackingRegions/plugins/AreaSeededTrackingRegionsBuilder.cc

```diff
@@ -228,29 +234,93 @@ std::unique_ptr<TrackingRegion> AreaSeededTrackingRegionsBuilder::Builder::regio
 
   const auto meanEta = (minEta+maxEta)/2.f;
   const auto meanPhi = (minPhi+maxPhi)/2.f;
-  const auto deltaEta = maxEta-meanEta + m_conf->m_extraEta;
-  const auto deltaPhi = maxPhi-meanPhi + m_conf->m_extraPhi;
+  const auto dEta = maxEta-meanEta + m_conf->m_extraEta;
+  const auto dPhi = maxPhi-meanPhi + m_conf->m_extraPhi;
+
+  int n_objects = 0;
+  if (candidates.first) n_objects = candidates.first->size();
+
+  if (n_objects > 0){
+	// If we have objected used for seeding, loop over objects and find overlap with the found region. Return overlaps as tracking regions to use
+	for(const auto& object : *candidates.first) {
+		float dEta_Cand = candidates.second.first;
+		float dPhi_Cand = candidates.second.second;
+	       	float eta_Cand = object.eta();
+		float phi_Cand = object.phi();
+          	float dEta_Cand_Point = std::abs(eta_Cand-meanEta);
+          	float dPhi_Cand_Point = std::abs(deltaPhi(phi_Cand,meanPhi));
+
+		if(dEta_Cand_Point > (dEta_Cand + dEta) || dPhi_Cand_Point > (dPhi_Cand + dPhi)) continue;
+
+		float  etaMin_RoI = std::max(eta_Cand-dEta_Cand,meanEta-dEta);
+		float  etaMax_RoI = std::min(eta_Cand+dEta_Cand,meanEta+dEta);
+
+		float  phi_Cand_minus  = normalizedPhi(phi_Cand-dPhi_Cand);
+		float  phi_Point_minus = normalizedPhi(meanPhi-dPhi);
+		float  phi_Cand_plus  = normalizedPhi(phi_Cand+dPhi_Cand);
+		float  phi_Point_plus = normalizedPhi(meanPhi+dPhi);
+
+		float phiMin_RoI = deltaPhi(phi_Cand_minus,phi_Point_minus)>0. ? phi_Cand_minus : phi_Point_minus ;	
+		float phiMax_RoI = deltaPhi(phi_Cand_plus,phi_Point_plus)<0. ? phi_Cand_plus : phi_Point_plus;
+
+
+		const auto meanEtaTemp = (etaMin_RoI+etaMax_RoI)/2.f;
+		auto meanPhiTemp = (phiMin_RoI+phiMax_RoI)/2.f;
+		if( phiMax_RoI < phiMin_RoI ) meanPhiTemp-=M_PI;
+	        meanPhiTemp = normalizedPhi(meanPhiTemp);
+	        
+		const auto dPhiTemp = deltaPhi(phiMax_RoI,meanPhiTemp);
+		const auto dEtaTemp = etaMax_RoI-meanEtaTemp;
+
+		const auto x = std::cos(meanPhiTemp);
+  	  	const auto y = std::sin(meanPhiTemp);
+  	  	const auto z = (x*x+y*y)/std::tan(2.f*std::atan(std::exp(-meanEtaTemp))); // simplify?
```

Comment:
```(x*x+y*y)``` is identically equal to 1 here: since you want to reduce the cpu cosumption in your algo, please avoid useless computations

Moreover, and in the same spirit: ```(x,y,z) ``` are only computed to be passed to the constructor of the GlobalVector a few lines below: did you check wheter by using the constructor with ```(eta, phi, r)```, that can avoid making those computations yourself, you can further optimize?

Reply:
Thanks for the comments. I can look into this, but before I change anything I would certainly like to hear @makortel  's views on this as that is in his part of the code.

Reply:
Sure, go a head and simplify.

In longer run one could also investigate if one could replace passing the direction with `GlobalVector` by just phi and eta (I suspect all uses of the direction inside the regions and their clients could very well use only phi and eta).

Reply:
Simplifying the calculation improve the module timing by 0.6%. I also tried using the different constructor, but that uses theta in the constructor (according to https://twiki.cern.ch/twiki/bin/view/CMSPublic/WorkBookChapter9#GlobalPoint_and_GlobalVector) so we still needed one of the calculation and it ended up being slightly less fast than keeping it in cartesian coordinates after all.